internal/resource/config/passwordcredentialvalidator/password_credential_validator_resource.go

```diff
@@ -122,6 +122,119 @@ func (r *passwordCredentialValidatorResource) Schema(ctx context.Context, req re
 	resp.Schema = schema
 }
 
+// Metadata returns the resource type name.
+func (r *passwordCredentialValidatorResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_password_credential_validator"
+}
+
+func (r *passwordCredentialValidatorResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func (r *passwordCredentialValidatorResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	var model passwordCredentialValidatorModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &model)...)
+
+	tableNames := []string{}
+	nestedTableFieldNameValues := []string{}
+	var rowCount int
+	configuration := model.Configuration.Attributes()
+	tables := configuration["tables"].(types.List).Elements()
+	for _, table := range tables {
+		tableAttrs := table.(types.Object).Attributes()
+		tableName := tableAttrs["name"].(types.String).ValueString()
+		tableNames = append(tableNames, tableName)
+		if tableName == "Users" {
+			tableRow := tableAttrs["rows"].(types.List).Elements()
+			rowCount = len(tableRow)
+			for _, row := range tableRow {
+				rowAttrs := row.(types.Object).Attributes()
+				fields := rowAttrs["fields"].(types.List).Elements()
+				for _, field := range fields {
+					fieldRow := field.(types.Object).Attributes()
+					nestedTableFieldName := fieldRow["name"].(types.String).ValueString()
+					nestedTableFieldNameValues = append(nestedTableFieldNameValues, nestedTableFieldName)
+				}
+			}
+		}
+	}
+
+	fieldNameValues := []string{}
+	fields := configuration["fields"].(types.List).Elements()
+	for _, field := range fields {
+		field := field.(types.Object).Attributes()
+		fieldName := field["name"].(types.String).ValueString()
+		fieldNameValues = append(fieldNameValues, fieldName)
+	}
+
+	pluginDescriptorRefId := model.PluginDescriptorRef.Attributes()["id"].(types.String).ValueString()
+	switch pluginDescriptorRefId {
+	case "com.pingconnect.alexandria.pingfed.pcv.PingOnePasswordValidator":
+		hasClientId, clientIdCount := internaltypes.ValidateStringCountInSlice(fieldNameValues, "Client Id", 1)
+		if !hasClientId || clientIdCount == 0 {
+			resp.Diagnostics.AddError("The \"Client Id\" field is required for the PingOne for Enterprise Directory Password Credential Validator", "")
```

Comment:
It seems like these errors should indicate exactly one value is expected. Since you would get this same error message if you had two Client Id values, if I'm understanding it right.

Reply:
you can only have one value of each in this one